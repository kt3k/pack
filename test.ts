import { assertEquals, assertStringIncludes } from "@std/assert"
import { stripAnsiCode } from "@std/fmt/colors"
import { assertSnapshot } from "@std/testing/snapshot"
import "@std/encoding";

const td = new TextDecoder()

Deno.test("cmd -h", async () => {
  const { stdout } = await cmd("-h")
  assertStringIncludes(stdout, "Usage: deno run -A jsr:@kt3k/pack")
})

Deno.test("cmd -v", async () => {
  const { stdout } = await cmd("-v")
  assertStringIncludes(stdout, "@kt3k/pack@")
})

Deno.test("cmd example.ts -o example-out.js", async (t) => {
  // Cache dependencies of hello.ts
  await deno("cache", "example.ts")
  const { stderr } = await cmd("example.ts", "-o", "example-out.js")
  const strippedStderr = stripAnsiCode(stderr)
  assertEquals(
    strippedStderr,
    `Bundling the file(s).
input: example.ts
output: example-out.js
`,
  )
  await assertSnapshot(t, await Deno.readTextFile("example-out.js"))
})

Deno.test("cmd example.ts", async (t) => {
  const { stdout, stderr } = await cmd("example.ts")
  const strippedStderr = stripAnsiCode(stderr)
  assertEquals(
    strippedStderr,
    `Bundling the file(s).
input: example.ts
output: STDOUT
`,
  )
  await assertSnapshot(t, await stdout)
})

Deno.test("cmd example-im.ts (deno.json detection)", async (t) => {
  const { stdout, stderr } = await cmd("example-im.ts")
  const strippedStderr = stripAnsiCode(stderr)
  assertEquals(
    strippedStderr,
    `Bundling the file(s).
input: example-im.ts
output: STDOUT
`,
  )
  await assertSnapshot(t, await stdout)
})

async function deno(...args: string[]) {
  const cmd = new Deno.Command("deno", { args })
  const output = await cmd.output()
  return {
    stdout: td.decode(output.stdout),
    stderr: td.decode(output.stderr),
  }
}

function cmd(...args: string[]) {
  return deno("run", "-A", "main.ts", ...args)
}
