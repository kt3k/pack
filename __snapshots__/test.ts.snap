export const snapshot = {};

snapshot[`cmd example.ts -o example-out.js 1`] = `
'// https://jsr.io/@std/encoding/0.223.0/_util.ts
var encoder = new TextEncoder();
function getTypeName(value) {
  const type = typeof value;
  if (type !== "object") {
    return type;
  } else if (value === null) {
    return "null";
  } else {
    return value?.constructor?.name ?? "object";
  }
}
function validateBinaryLike(source) {
  if (typeof source === "string") {
    return encoder.encode(source);
  } else if (source instanceof Uint8Array) {
    return source;
  } else if (source instanceof ArrayBuffer) {
    return new Uint8Array(source);
  }
  throw new TypeError(
    \`The input must be a Uint8Array, a string, or an ArrayBuffer. Received a value of the type \${getTypeName(source)}.\`
  );
}

// https://jsr.io/@std/encoding/0.223.0/hex.ts
var hexTable = new TextEncoder().encode("0123456789abcdef");
var textEncoder = new TextEncoder();
var textDecoder = new TextDecoder();
function encodeHex(src) {
  const u8 = validateBinaryLike(src);
  const dst = new Uint8Array(u8.length * 2);
  for (let i = 0; i < dst.length; i++) {
    const v = u8[i];
    dst[i * 2] = hexTable[v >> 4];
    dst[i * 2 + 1] = hexTable[v & 15];
  }
  return textDecoder.decode(dst);
}

// example.ts
console.log(encodeHex(Uint8Array.from([0, 1, 2, 3])));
'
`;

snapshot[`cmd example.ts 1`] = `
'// https://jsr.io/@std/encoding/0.223.0/_util.ts
var encoder = new TextEncoder();
function getTypeName(value) {
  const type = typeof value;
  if (type !== "object") {
    return type;
  } else if (value === null) {
    return "null";
  } else {
    return value?.constructor?.name ?? "object";
  }
}
function validateBinaryLike(source) {
  if (typeof source === "string") {
    return encoder.encode(source);
  } else if (source instanceof Uint8Array) {
    return source;
  } else if (source instanceof ArrayBuffer) {
    return new Uint8Array(source);
  }
  throw new TypeError(
    \`The input must be a Uint8Array, a string, or an ArrayBuffer. Received a value of the type \${getTypeName(source)}.\`
  );
}

// https://jsr.io/@std/encoding/0.223.0/hex.ts
var hexTable = new TextEncoder().encode("0123456789abcdef");
var textEncoder = new TextEncoder();
var textDecoder = new TextDecoder();
function encodeHex(src) {
  const u8 = validateBinaryLike(src);
  const dst = new Uint8Array(u8.length * 2);
  for (let i = 0; i < dst.length; i++) {
    const v = u8[i];
    dst[i * 2] = hexTable[v >> 4];
    dst[i * 2 + 1] = hexTable[v & 15];
  }
  return textDecoder.decode(dst);
}

// example.ts
console.log(encodeHex(Uint8Array.from([0, 1, 2, 3])));
'
`;

snapshot[`cmd example-im.ts (deno.json detection) 1`] = `
'// https://jsr.io/@std/encoding/0.224.3/_validate_binary_like.ts
var encoder = new TextEncoder();
function getTypeName(value) {
  const type = typeof value;
  if (type !== "object") {
    return type;
  } else if (value === null) {
    return "null";
  } else {
    return value?.constructor?.name ?? "object";
  }
}
function validateBinaryLike(source) {
  if (typeof source === "string") {
    return encoder.encode(source);
  } else if (source instanceof Uint8Array) {
    return source;
  } else if (source instanceof ArrayBuffer) {
    return new Uint8Array(source);
  }
  throw new TypeError(
    \`The input must be a Uint8Array, a string, or an ArrayBuffer. Received a value of the type \${getTypeName(source)}.\`
  );
}

// https://jsr.io/@std/encoding/0.224.3/base32.ts
var lookup = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567".split("");
var revLookup = [];
lookup.forEach((c, i) => revLookup[c.charCodeAt(0)] = i);

// https://jsr.io/@std/encoding/0.224.3/base58.ts
var base58alphabet = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz".split("");

// https://jsr.io/@std/encoding/0.224.3/hex.ts
var hexTable = new TextEncoder().encode("0123456789abcdef");
var textEncoder = new TextEncoder();
var textDecoder = new TextDecoder();
function encodeHex(src) {
  const u8 = validateBinaryLike(src);
  const dst = new Uint8Array(u8.length * 2);
  for (let i = 0; i < dst.length; i++) {
    const v = u8[i];
    dst[i * 2] = hexTable[v >> 4];
    dst[i * 2 + 1] = hexTable[v & 15];
  }
  return textDecoder.decode(dst);
}

// https://jsr.io/@std/encoding/0.224.3/varint.ts
var AB = new ArrayBuffer(8);
var U32_VIEW = new Uint32Array(AB);
var U64_VIEW = new BigUint64Array(AB);

// example-im.ts
console.log(encodeHex(Uint8Array.from([0, 1, 2, 3])));
'
`;

snapshot[`cmd example-vscode.ts (deno.json detection) 1`] = `
'var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// example-vscode.ts
var example_vscode_exports = {};
__export(example_vscode_exports, {
  activate: () => activate
});
module.exports = __toCommonJS(example_vscode_exports);
var vscode = __toESM(require("vscode"));
function activate(context) {
  context.subscriptions.push(vscode.window.registerCustomEditorProvider(
    "kt3k.bwBlock",
    new BlockEdit(context)
  ));
}
var BlockEdit = class {
  constructor(context) {
    this.context = context;
  }
};
'
`;
